### PR TITLE
Remove redirect to previous page after manual log out, expose redirect param, fix auth page titles

### LINKED ISSUE #12311: Don't go back to the previous workspace when logging in with a different user
We currently try to bring the user back after login to where they cam from. This leads to the weird scenario, if a user will log out and sign in (or register) with a different user, that doesn't have permission to the same workspace, they'll just see an error message because they can't access that workspace. Despite us needing to improve that error message (tracked via https://github.com/airbytehq/airbyte/issues/10735) we should not try to bring you back to the same workspace if you change your user after going to the login screen, but instead trying to bring them to their default workspace instead.

### PR DESCRIPTION
**Pull request by edmundito**
## What
Fixes #12311 

This change fixes a number of issues related to logging in and out:

1. When logging out manually, the app will no longer provide a redirect location so that when the user logs in it go back to the app's home
2. The login, sign in, and reset password pages now have titles so that when navigating the title on the window updates
3. The "from" parameter on the log in page is now part of the page instead of an internal state

## How
For manually logging out, the auth service now holds a flag called `loggedOut` to indicate that the user-initiated the logout. It is reset when the user logs in.

The "from" parameter used to be part of react-router's state prop. Now it's just built into the navigation query and extracted from the query.

## Testing
- Sign out from settings and sign in
- Clear cookies, reload page, and sign in
- Navigate to reset password, sign up, and sign in

### CODE REVIEW

**Review by teallarson — CHANGES_REQUESTED** [review]
Tested locally.  When I log out and then register a new user, the app tries to navigate to the old user's workspace and I see `Unknown error occurred` in the UI.

File: airbyte-webapp/src/packages/cloud/views/auth/LoginPage/LoginPage.tsx [hunk 1]
```diff
@@ -35,18 +37,15 @@ const LoginPage: React.FC = () => {
...
-              .catch((err) => {
-                if (err instanceof FieldError) {
-                  setFieldError(err.field, err.message);
-                } else {
-                  setFieldError("password", err.message);
-                }
-              })
-          );
+          return login(values)
+            .then((_) => replace(query.from ?? "/"))
```
Comment (krishnaglick) [inline airbyte-webapp/src/packages/cloud/views/auth/LoginPage/LoginPage.tsx:41]:
```suggestion
            .then(() => replace(query.from ?? "/"))
```

**Review by teallarson — APPROVED** [review]
Registering a new user after logging out is fixed.  All other workflows listed in the PR work as well.

**Review by krishnaglick — APPROVED** [review]
Working locally for me 👍

**Pull request merged**